src/android/BuiltinUserCache.java

```diff
@@ -380,9 +381,74 @@ public JSONObject getLocalStorage(String key, boolean withMetadata) throws JSONE
         }
     }
 
+    /**
+     * Gets all data in DB given a key and returns it in a JSONArray
+     * @param key
+     * @param withMetadata
+     * @return JSONArray of data from SQL database
+     * @throws JSONException
+     */
+    public JSONArray getLocalStorageArr(String key, boolean withMetadata) throws JSONException {
+        JSONArray wrappedArray = wrapJson(getValuesForInterval(key, LOCAL_STORAGE_TYPE, getAllTimeQuery(cachedCtx), withMetadata), withMetadata);
+        if (wrappedArray.length() == 0) {
+            return new JSONArray();
+        } else {
+            return wrappedArray;
+        }
+    }
+    
+    /**
+     * Delete entries at a given key older than a certain time stamp. Used to bound the number of entries
+     * allowed in the DB at a given time.
+     * @param key
+     * @param writeTs
+     */
+    public void deleteEntriesOlderThan(String key, double writeTs){
```

Comment:
I know this is what I suggested, but on reflection, I think we should also add the type here to be safer. We don't re-use keys between types but if we did, this would delete entries in both (say) SENSOR_DATA and LOCAL_STORAGE

Reply:
Just fixed it, the new where clause is...

`String whereClause = KEY_KEY + " = ? AND " + KEY_WRITE_TS + " < " + writeTs + " AND " + KEY_TYPE + " = '" + LOCAL_STORAGE_TYPE + "'";`